### PR TITLE
Missing Core Ordering for tagged elements menu-item

### PR DESCRIPTION
**Pull request by LivioCavallo**
### Summary of Changes
Added "c.core_ordering" ordering criteria in com_tags configuration page.
Added "c.core_ordering" ordering criteria for tagged-elements menu-item type (com_tags)
Added text for tagged elements core.ordering criteria.

### Testing Instructions
Tagged elements menu-item offers some criteria to order elements (title, creation, publishing, ...) but that menu-item type is missing the core-ordering criteria.
Core-ordering would allow elements to mantaing the ordering specified for emelents own component.
For instance it would be useful to set the desired ordering for some tagged contacts in the backend and than have tagged contacts shown in the same order in your page.
This is core.ordering.

### Expected result
tagged elements ordered as in 'core-component' using core-ordering

### Actual result
core.ordering not available

### Documentation Changes Required
Simply add the proposed 3 lines in that 3 files and core-ordering will be available

### CODE REVIEW

**Comment by brianteeman** [comment]
Unfortunately this will not work and that is why it is not present in the core. 

The reason is that you can apply a tag to any supported component. So for example when you create a menu item for the "red" tag that include articles and contacts and there is no way for you to order them

**Comment by LivioCavallo** [comment]
Right, not a complete feature, but it will work in a single tag, single component scenario and in my opinion it would be useful anyway (as this way you can use tags as a replacement of the missing multi-category functionality and have a minimal per-component single-tag/category elements ordering).

**Comment by brianteeman** [comment]
and how do you explain the limitation to the user - otherwise they will think its a bug. This is a fundamental limitation of a vertical component such as tags

**Comment by LivioCavallo** [comment]
"core.ordering will work just with a single tag in a single component".

I agree with you: tags component should be expanded.
To improve ordering I suggest:

1) create a table #__tags_groups: id, tags_ids_listing (a key from #__tags.id table).
This will list only the tags groupings created by the user.

2) create a table #__tagged_groups_ordering: id, group-id (from #__tags_groups.id), ordering

3) expand component_tags to allow creation of tags_groups, listing of emelents with that tags and ordering of that elements.


This will be enought to support ordering of any combination of tags, across different components. So we'll have a real and complete core.ordering criteria.

Unfortunately at the moment I am not able to write 3)... but It seems to me that it should be quite standard code for people able to create components (I am not sure it's so easy considering tags hierarchy), so I hope that someone sooner or later will write it!

File: administrator/language/en-GB/en-GB.com_tags.ini [hunk 1]
```diff
@@ -187,3 +187,4 @@ COM_TAGS_TAGGED_ITEMS_LANGUAGE="Language"
 COM_TAGS_TAGGED_ITEMS_TITLE="Title"
 COM_TAGS_XML_DESCRIPTION="This component manages tags."
 JGLOBAL_NO_ITEM_SELECTED="No tags selected"
+JGLOBAL_CORE_ORDERING="Elements order (core.ordering)"
```
Comment (Quy) [inline administrator/language/en-GB/en-GB.com_tags.ini:190]: Alpha order. Move above line 189.

**Comment by brianteeman** [comment]
@mbabker can we have a decision on this PR please. 

https://issues.joomla.org/tracker/joomla-cms/17379#event-300520<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/17379">issues.joomla.org/tracker/joomla-cms/17379</a>.</sub>

**Comment by mbabker** [comment]
I think if we're going to add the ordering option we really need to make it work consistently with cross-component data (and as pointed out, that's not easy; I had to do something recently to allow a client to order featured items for 3 different data types and that's a lot of mapping information to store for such an edge case feature).

I don't think we should merge this with the limitation pointed out.

**Comment by brianteeman** [comment]
OK - thanks. I am therefore closing this at this time.

@LivioCavallo if you can find a way to make it work cross-component then you can always create a new PR or updaate this one and it can be reopened. Thanks for trying

**Pull request closed without merging**